sky/provision/kubernetes/volume.py

```diff
@@ -45,17 +46,26 @@ def check_pvc_usage_for_pod(context: Optional[str], namespace: str,
         access_mode = pvc.spec.access_modes[0]
         if access_mode not in once_modes:
             continue
-        usedby = _get_volume_usedby(context, namespace, pvc_name)
-        if usedby:
+        usedby_pods, _ = _get_volume_usedby(context, namespace, pvc_name)
+        if usedby_pods:
             raise config_lib.KubernetesError(f'Volume {pvc_name} with access '
                                              f'mode {access_mode} is already '
-                                             f'in use by {usedby}.')
+                                             f'in use by Pods {usedby_pods}.')
 
 
 def apply_volume(config: models.VolumeConfig) -> models.VolumeConfig:
     """Creates or registers a volume."""
     context, namespace = _get_context_namespace(config)
     pvc_spec = _get_pvc_spec(namespace, config)
+    # Check if the storage class exists
+    storage_class_name = pvc_spec['spec'].get('storageClassName')
+    if storage_class_name is not None:
+        try:
+            kubernetes.storage_api(context).read_storage_class(
+                name=storage_class_name)
```

Comment:
This increases the permission scope required by SkyPilot. We may need to add to our minimal kubernetes permissions doc and the roles created by helm chart